airflow/jobs.py

```diff
@@ -1701,6 +1701,16 @@ def _update_counters(self, started, succeeded, skipped, failed, tasks_to_run):
                                     .format(ti))
                 started.pop(key)
                 tasks_to_run[key] = ti
+            # special case: if we hit concurrency limits put it back in the queue
+            elif ti.state == State.NONE:
+                self.logger.warning("FIXME: task instance {} state was set to "
```

Comment:
This is from a past discussion with @saguziel and @aoen where we decided not to introduce another task state (CONCURRENCY_REACHED). I didn't like that cause it makes NONE ambiguous so hence the fix me. It mirrors the log message in TaskInstance. I will amend the comment.